src/util/color.h

```diff
@@ -2,51 +2,264 @@
+            return QObject::tr("Red");
+        } else if (color.name().toUpper() == Green.toUpper()) {
+            return QObject::tr("Green");
+        } else if (color.name().toUpper() == Yellow.toUpper()) {
+            return QObject::tr("Yellow");
+        } else if (color.name().toUpper() == Blue.toUpper()) {
+            return QObject::tr("Blue");
+        } else if (color.name().toUpper() == Orange.toUpper()) {
+            return QObject::tr("Orange");
+        } else if (color.name().toUpper() == Purple.toUpper()) {
+            return QObject::tr("Purple");
+        } else if (color.name().toUpper() == Cyan.toUpper()) {
+            return QObject::tr("Cyan");
+        } else if (color.name().toUpper() == Magenta.toUpper()) {
+            return QObject::tr("Magenta");
+        } else if (color.name().toUpper() == Lime.toUpper()) {
+            return QObject::tr("Lime");
+        } else if (color.name().toUpper() == Pink.toUpper()) {
+            return QObject::tr("Pink");
+        } else if (color.name().toUpper() == Teal.toUpper()) {
+            return QObject::tr("Teal");
+        } else if (color.name().toUpper() == Lavender.toUpper()) {
+            return QObject::tr("Lavender");
+        } else if (color.name().toUpper() == Brown.toUpper()) {
+            return QObject::tr("Brown");
+        } else if (color.name().toUpper() == Beige.toUpper()) {
+            return QObject::tr("Beige");
+        } else if (color.name().toUpper() == Maroon.toUpper()) {
+            return QObject::tr("Maroon");
+        } else if (color.name().toUpper() == Mint.toUpper()) {
+            return QObject::tr("Mint");
+        } else if (color.name().toUpper() == Olive.toUpper()) {
+            return QObject::tr("Olive");
+        } else if (color.name().toUpper() == Apricot.toUpper()) {
+            return QObject::tr("Apricot");
+        } else if (color.name().toUpper() == Navy.toUpper()) {
+            return QObject::tr("Navy");
+        } else if (color.name().toUpper() == Grey.toUpper()) {
+            return QObject::tr("Grey");
+        } else if (color.name().toUpper() == White.toUpper()) {
+            return QObject::tr("White");
+        } else if (color.name().toUpper() == Black.toUpper()) {
+            return QObject::tr("Black");
+        }
+        return QObject::tr("Undefined Color");
+    };
 
-// if the ColorToChooseBy is darker than the global threshold,
-// the Color from the second argument will be returned.
+    // Returns a new default colors representation, i.e. maps each default color to itself.
+    // Stores the color's name() property, e.g. "#A9A9A9"
+    static std::unique_ptr<ColorsRepresentation> defaultRepresentation() {
+        std::unique_ptr<ColorsRepresentation> representation = std::make_unique<ColorsRepresentation>();
+        for (QColor color : predefinedColors()) {
+            representation->setRepresentation(color, color);
+        }
+        return representation;
```

Comment:
I think we need here std::move(representation) for some compilers.

Reply:
If I move here I get this warning:
`warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]`